### PR TITLE
Add benchmarks for parameter resolution

### PR DESCRIPTION
**Pull request by tanujkhattar**
Part of https://github.com/quantumlib/Cirq/issues/3840
cc https://github.com/qh-lab/pyle/issues/29309

The benchmarks produce the following output:

```
              --                                     num_scan_points                          
              ------------ -------------------------------------------------------------------
               num_qubits        20            40           60           80           100     
              ============ ============= ============= ============ ============ =============
                   50       18.9±0.04ms   37.8±0.06ms   57.4±0.9ms   75.7±0.1ms   94.0±0.09ms 
                  100        36.4±0.2ms    72.8±0.1ms   109±0.2ms    145±0.1ms      183±1ms   
                  150        53.6±0.1ms    107±0.2ms    162±0.2ms     215±1ms       267±1ms   
                  200        70.5±0.2ms     141±1ms      213±1ms      284±2ms      352±0.5ms  
              ============ ============= ============= ============ ============ =============

```

### CODE REVIEW

File: check/asv_run [hunk 1]
```diff
@@ -11,5 +11,4 @@
 cd "$(dirname "${BASH_SOURCE[0]}")"
 cd "$(git rev-parse --show-toplevel)"
 
-export ASV_PYTHONPATH="$(git rev-parse --show-toplevel)/examples"
-asv run "$@"
+asv run "$@"
```
Comment (pavoljuhas) [inline check/asv_run:14]: Nit - text files should end with a newline character.

File: benchmarks/parameter_resolution.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import cirq
+import sympy
+import random
+import numpy as np
```
Comment (pavoljuhas) [inline benchmarks/parameter_resolution.py:18]: Let's fix import order to standard-third_party-this_package, ie, [random, numpy, sympy, cirq].

**Review by pavoljuhas — APPROVED** [review]
LGTM after formatting fixups.

File: check/asv_run (hunk 1 above)
Comment (tanujkhattar) [reply check/asv_run]: done

File: benchmarks/parameter_resolution.py (hunk 2 above)
Comment (tanujkhattar) [reply benchmarks/parameter_resolution.py]: done

**Review by pavoljuhas — APPROVED** [review]
LGTM

**Pull request merged**